Client/Frontend/Settings/SearchCustomEngineViewController.swift

```diff
@@ -493,23 +495,38 @@ extension SearchCustomEngineViewController: UITextViewDelegate {
             return false
         }
         
-        if let copiedText = UIPasteboard.general.string, text.contains(copiedText) {
-            guard copiedText.hasPrefix("https://") else {
-                handleError(error: SearchEngineError.insecureURL)
-                return false
-            }
-            
-            return true
-        } else {
-            let textLengthInRange = textView.text.count + (text.count - range.length)
+        let textLengthInRange = textView.text.count + (text.count - range.length)
             
-            // The default text "https://" cant ne deleted or changed so nothing without a secure scheme can be added
-            return textLengthInRange <= Constants.urlEntryMaxCharacterCount && textLengthInRange >= 8
-        }
+        // The default text "https://" cant ne deleted or changed so nothing without a secure scheme can be added
+        return textLengthInRange <= Constants.urlEntryMaxCharacterCount && textLengthInRange >= 8
     }
     
     func textViewDidChange(_ textView: UITextView) {
         changeAddButton(for: .disabled)
+        
+        // This conditional clause is added in order to force use to use secure url scheme
+        // Instead of checking paste-board with every character entry, the textView text is analyzed
+        // And according to what prefix the copied or entered text has we alter the result to start with https://
+        // Also block repeating https:// and http:// schemes
+        if !textView.text.hasPrefix("https://") {
```

Comment:
Can we move this logic out and write a unit test for it?
If you feel like this code is universal enough you can put it in String extensions,
if not, just extract it to a method in this class and write a test for it

Reply:
great idea ! lets do it

Reply:
Logic is separated and test is added.
Fix 7e10a68